src/App.tsx

```diff
@@ -1,18 +1,37 @@
-import React from 'react';
+import React, { ChangeEvent, useState } from 'react';
 import './App.css';
+import { Pagination } from './components/Pagination';
 import { getNumbers } from './utils';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42)
-  .map(n => `Item ${n}`);
-
 export const App: React.FC = () => {
+  const totalItems = 42;
+  const firstPage = 1;
+
+  const [itemsOnPage, setItemsOnPage] = useState(5);
+  const [initialPage, setInitialPage] = useState(firstPage);
+
+  const maxItems = initialPage * itemsOnPage;
+  const startVisibleItems = maxItems - itemsOnPage;
+  const endVisibleItems = maxItems > totalItems
+    ? totalItems
+    : maxItems;
+
+  const selectedItems = (
+    event: ChangeEvent<HTMLSelectElement>,
+  ) => {
+    setItemsOnPage(+event.target.value);
+  };
+
+  const changedPage = (page: number) => {
+    setInitialPage(page);
+  };
```

Comment:
Do you really need to wrap it into a function? You can directly use setInitialPage I suppose